src/Microsoft.AspNetCore.Blazor/Rendering/ComponentState.cs

```diff
@@ -96,5 +96,11 @@ public void DispatchEvent(EventHandlerInvoker binding, UIEventArgs eventArgs)
 
         public void NotifyRenderCompleted()
             => (_component as IHandleAfterRender)?.OnAfterRender();
+
+        public void SetDirectParameters(ParameterCollection parameters)
+        {
+            // TODO: Attach tree parameters also
+            Component.SetParameters(parameters);
```

Comment:
I'm a little confused by this because it doesn't seem like tree parameters belong in `SetDirectParameters`

Reply:
OK DOPE